### DISCUSSION TITLE
Data  not found in NOAA website.

### DISCUSSION THREAD

**Discussion by oye-bobs** [cu-esiil-edu/2024-data-short-course#56] [Ideas]
@eculler @nquarder In an attempt to retry the project by myself, I tried to use the search button to look for daily summaries data, However I was totally dissapointed when after editing the api url link, the data for PRCP and TOBS were Blank.

How can I work around it?
And How can I import the api url for a new project?

**Comment by nquarder** [comment]
Hi @oye-bobs. I was able to find very limited data for 10 site locations in Nigeria using the Climate Data Online Search tool: https://www.ncdc.noaa.gov/cdo-web/search

You can look for **Countries** using the drop down menu under **Search For**. I typed in Nigeria and got back 10 stations but with very limited coverage: https://www.ncdc.noaa.gov/cdo-web/datasets/GHCND/locations/FIPS:NI/detail (click on 'See station list below)

It looks like this station at **Murtala Muhammed** has data going back to the 1970's but has pretty spotty coverage: 
https://www.ncdc.noaa.gov/cdo-web/datasets/GHCND/stations/GHCND:NIM00065201/detail

Maybe we can find some open weather data for Nigerian stations using a more local service?

**Comment by nquarder** [comment]
Hey again @oye-bobs, I didn't see the last part of your question about changing the API to look at a new location for another project. 

You probably noticed near the top of the notebook where we create a API endpoint request for the weather station in Boulder, CO. 

`ncei_api_url = (
'https://www.ncei.noaa.gov/access/services/data/v1?dataset=daily-summaries&dataTypes=TOBS,PRCP&stations=USC00050848&startDate=1893-10-01&endDate=2024-02-18&includeStationName=true&includeStationLocation=1&units=standard')`

We can use that same request, with some minor edits, to look at other locations, time periods, and weather variables. 

For example, if we wanted to use that station from Murtala Muhammed that I linked to above (network ID `GHCND:NIM00065201`) we would change our `stations=` parameter in the API call from `USC00050848` to `NIM00065201`. 

Likewise, we could change the `startDate=` parameter from `1893-10-01` to `1973-02-06` since that is when data collection began at the Murtala Muhammed location. Looking into the site details a bit deeper it appears this station at Murtala Muhammed has values `TMAX`, `TMIN`, and `TAVG` but no `TOBS`. You would want to make this change with the `dataTypes=` parameter.

If you want to read more about how create an endpoint request using the NOAA/NCEI API here: https://www.ncei.noaa.gov/support/access-data-service-api-user-documentation

**Reply by oye-bobs** [reply]
@nquarder This is exactly what I did while running my code, but when I tried to display the Dataframe, only (NaN) Values were present in the "TOBS and PRCP" columns.

**Reply by nquarder** [reply]
Okay @oye-bobs did you try replacing `TOBS` with `TAVG`, `TMIN`, `TMAX` ? There is no `TOBS` for the Murtala station. 
![Screen Shot 2024-04-19 at 8 36 49 AM](https://github.com/cu-esiil-edu/2024-data-short-course/assets/16786791/a3b1e912-6c3f-4fa0-8a89-5d19ef79f10d)

Also the `PRCP` data appears to have only 5% coverage going back to 1974 so I would expect it to be mostly NaN. 

![Screen Shot 2024-04-19 at 8 36 24 AM](https://github.com/cu-esiil-edu/2024-data-short-course/assets/16786791/f1397247-2e2e-45e1-8a53-c72c59c37a1a)

You can try plotting the precipitation data to see where the coverage is but I suspect it won't be enough to do meaningful analysis. Give this a shot and let us know what you come up with! [**EDIT:** fairly decent coverage in the precip data from 2000-2020]

**Reply by nquarder** [reply]
@oye-bobs It looks like there is pretty good coverage since about the year 2000 but before then the data are fairly inconsistent. This is the average temperature (TAVG).
![image](https://github.com/cu-esiil-edu/2024-data-short-course/assets/16786791/b67e6681-570e-4d00-8258-a4b7844007a6)

**Comment by oye-bobs** [comment]
@nquarder Alright, I would try again later.

**Comment by oye-bobs** [comment]
@nquarder I tried to run it again, and got an error. Help me inspect if the API is correct
![Screenshot 2024-04-19 214248](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/f2b8116e-b833-4395-b537-269ebc563eb6)

**Reply by oye-bobs** [reply]
Although the data was displayed, It appears 'TAVG', 'TMAX', and 'TMIN' are absent.

**Reply by oye-bobs** [reply]
Here's what I mean.
![Screenshot 2024-04-19 214531](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/4bba26f0-d0de-49b2-9dc0-7bb73e631710)

**Reply by nquarder** [reply]
Hi @oye-bobs - it looks like you still have `TOBS` in your API call. Can you change that to `TAVG, TMAX, TMIN` and then try re-running the cell?

My API for this station looks like this:

`https://www.ncei.noaa.gov/access/services/data/v1?dataset=daily-summaries&dataTypes=TAVG,TMIN,TMAX,PRCP&stations=NIM00065201&startDate=1973-02-06&endDate=2024-02-18&includeStationName=true&includeStationLocation=1&units=standard`

**Reply by oye-bobs** [reply]
@nquarder Oh, I totally didn't realise. Thank you.

**Reply by oye-bobs** [reply]
@nquarder I've run into another error. Apparently, after running this code block,
![1](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/f9c52634-1230-4049-ae33-3776b9d343e2)


I got this error
![2](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/b28ea335-0ba6-42fa-8911-477e78f8b08d)

**Reply by nquarder** [reply]
@oye-bobs without being able to see your API call I would suspect that you are missing `TMIN`, `TMAX` from your `dataTypes=` parameter. 

If you want to be able to look at precipitation as well as average, minimum, and maximum temperature, you'll want to make sure that you have the following in your API call

`dataTypes=PRCP,TAVG,TMIN,TMAX`

**Reply by oye-bobs** [reply]
This is my API call though.
![qq](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/ceee21a3-1a29-4cfc-919f-47136ac83a0b)

**Reply by nquarder** [reply]
Hmmm that is weird @oye-bobs. I'm wondering what does your `murtala_df` look like when you open it up using pandas, but _before_ running the step where you clean up the data frame?

What does your data look like after you run this step?: 

`# Open data with pandas`
`murtala_df = pd.read_csv(ncei_api_url,
  index_col='DATE',
  parse_dates=True,
  na_values=['NaN'])`

`# Display Dataframe Data`
`murtala_df`

**Reply by oye-bobs** [reply]
This is what the output looks like.
![3](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/e05ed636-dbda-4dab-8008-cdfa84d2dd34)

**Comment by nquarder** [comment]
Aha! I think I see now the issue @oye-bobs :eyes:

I'm looking at your notebook now (https://github.com/cu-esiil-edu/shortcourse-01-get-started-oye-bobs/blob/main/new_notebook_for_reproducible_science.ipynb) and I see an extra empty space between `murtala_df` and the first square bracket in the code you're using to clean up the data frame:

`# Clean up the datafrae`
`murtala_df = murtala_df [['PRCP', 'TAVG','TMIN','TMAX']]`
`murtala_df`

I think you want this: 
`murtala_df = murtala_df[['PRCP', 'TAVG','TMIN','TMAX']]`

**Reply by oye-bobs** [reply]
Alright, the error has been resolved now. Thank you @nquarder

**Reply by nquarder** [reply]
nice work @oye-bobs !
